src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -118,7 +118,7 @@ class RefreshIndexTests extends QueryTest with HyperspaceSuite {
 
       val ex = intercept[HyperspaceException](hyperspace.refreshIndex(indexConfig.indexName))
       assert(
-        ex.getMessage.contains(s"Index refresh (to handle deleted source data) is " +
+        ex.getMessage.contains(s"Index refresh (to handle deleted or appended source data) is " +
           "only supported on an index with lineage."))
```

Comment:
Sorry, I'm confused (again!). Why do we need lineage for handling appends?

Reply:
We dont need lineage for appended. Lineage is needed for deleted source data files. However, both `excluded` and `appended` files are updated simultaneously and under same call to `RefreshLogEntryAction`. That is why lineage column check happens for such an action call and this test checks that.

Reply:
Thanks, that's what I thought. In that case, is there a way this message can be improved? If let's say something failed when handling appended source data (hypothetically), saying that we do not support this over an index with no lineage doesn't seem proper.

Reply:
I changed the exception msg to:
`Index refresh to update source content in index metadata is only supported on an index with lineage.`.
This still has an implicit restriction that index has to have lineage so we one can trigger an action to update index metadata.
The reason for that is currently "delete on read" is the only scenario that is relying/using such an index metadata update.
We can lift this restriction and distinguish between a case where we only have "appended" files once the usage of that update is added.